tests/integrations/raftstore/test_conf_change.rs

```diff
@@ -863,3 +863,34 @@ fn test_conf_change_fast() {
     must_get_equal(&cluster.get_engine(2), b"k1", b"v1");
     assert!(timer.saturating_elapsed() < Duration::from_secs(5));
 }
+
+#[test_case(test_raftstore::new_node_cluster)]
+#[test_case(test_raftstore::new_server_cluster)]
+fn test_remove_node_on_partition() {
+    let count = 6;
+    let mut cluster = new_cluster(0, count);
+    let pd_client = Arc::clone(&cluster.pd_client);
+    // Disable default max peer number check.
+    pd_client.disable_default_operator();
+    cluster.cfg.raft_store.raft_heartbeat_ticks = 1;
+    cluster.cfg.raft_store.raft_base_tick_interval = ReadableDuration::millis(10);
+    let r1 = cluster.run_conf_change();
+
+    cluster.must_put(b"k0", b"v0");
+    pd_client.must_add_peer(r1, new_peer(2, 2));
+    must_get_equal(&cluster.get_engine(2), b"k0", b"v0");
+    pd_client.must_add_peer(r1, new_peer(3, 3));
+    must_get_equal(&cluster.get_engine(3), b"k0", b"v0");
+
+    // peer 3 isolation
+    cluster.add_send_filter(IsolationFilterFactory::new(3));
+    // sleep for 10 heartbeat interval
+    thread::sleep(cluster.cfg.raft_store.raft_heartbeat_interval() * 10);
+    pd_client.remove_peer(r1, new_peer(2, 2));
+    thread::sleep(Duration::from_millis(500));
```

Comment:
```suggestion
    pd_client.remove_peer(r1, new_peer(2, 2));
    thread::sleep(Duration::from_millis(500));
    cluster.must_put(b"k1", b"v1");
```

Make sure region is available after every conf change.